### PR TITLE
Upstream changes (#144)

### PR DESCRIPTION
**Pull request by JayH5**
docker-library/python#144:
- Add "--force-reinstall" and a comment as to why to our "pip install" lines (docker-library/python@855b85c)

### CODE REVIEW

File: 2.7/Dockerfile [hunk 1]
```diff
@@ -46,7 +46,12 @@ RUN set -ex \
...
             && python2 /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION" \
             && rm /tmp/get-pip.py \
-        && pip install --no-cache-dir --upgrade pip==$PYTHON_PIP_VERSION \
+# we use "--force-reinstall" for the case where the version of pip we're trying to install is the same as the version bundled with Python
+# ("Requirement already up-to-date: pip==8.1.2 in /usr/local/lib/python3.6/site-packages")
+# https://github.com/docker-library/python/pull/143#issuecomment-241032683
+        && pip install --no-cache-dir --upgrade --force-reinstall "pip==$PYTHON_PIP_VERSION" \
+# then we use "pip list" to ensure we don't have more than one pip version installed
+# https://github.com/docker-library/python/pull/100
         && [ "$(pip list |tac|tac| awk -F '[ ()]+' '$1 == "pip" { print $2; exit }')" = "$PYTHON_PIP_VERSION" ] \
```
Comment (jerith) [inline 2.7/Dockerfile:55]: Are comments between line-continuations legal in dockerfiles? I know they don't work in bash and various other places. (The first line of the comment is seen as the continued line and then whatever comes afterwards is seen as a new line.)

Comment (JayH5) [inline 2.7/Dockerfile:55]: It seems they are...
Last I checked the Dockerfile parser was very simplistic (although it seems to have got slightly fancier in 1.12). It basically just checks for a `#` at the start of a line (the only kind of supported comment) and ignores that line.

Comment (jerith) [inline 2.7/Dockerfile:55]: Most things only do the comment ignoring after joining up continued lines or something. If you've tested that this works I'm happy with it, though.

**Comment by jerith** [comment]
:+1:

**Pull request merged**